code/game/machinery/doors/door.dm

```diff
@@ -128,6 +135,8 @@
 		user = null
 
 	if(density && !(obj_flags & EMAGGED))
+		if (check_locked(user))
+			return
 		if(allowed(user))
 			open()
 		else
```

Comment:
|| (locked)) , again, more consistent behavior and setting up to cut out a number of lines of code out of airlocks.dm in the future

Or vice-versa and moving a lot of this code into airlocks.dm